### PR TITLE
[DB-304][risk=no] Add Configurable API Test

### PR DESCRIPTION
**Pull request by rushtong**
Adds to https://precisionmedicineinitiative.atlassian.net/browse/DB-304, specifically the ability to generalize the configuration a little more.

### CODE REVIEW

File: performance/src/gatling/simulations/org/pmiops/databrowser/Scenarios.scala [hunk 1]
```diff
@@ -3,24 +3,48 @@ package org.pmiops.databrowser
...
 case object HomePage extends UserScenario { val name = "Home Page" }
 case object UserStorySmoking extends UserScenario { val name = "User Story: Smoking" }
 
 /**
   * User scenarios are defined here. Scenarios can be comprised of different Pages.
+  *
+  * Scenario names must be unique which is why there is a bit of name duplication.
+  * In the case where we want to run the same scenario under multiple conditions,
+  * (i.e. different #s of users across different ramp up times) we have to give
+  * each condition a unique name, but also need the s
```
Comment (calbach) [inline performance/src/gatling/simulations/org/pmiops/databrowser/Scenarios.scala]: Was this comment finished?

File: performance/src/gatling/simulations/org/pmiops/databrowser/Scenarios.scala [hunk 2]
```diff
@@ -3,24 +3,48 @@ package org.pmiops.databrowser
...
+    Configuration.userRampUpTimes.map { r =>
+      val name = s"Domain Search: $t: users: ${r._1}; time: ${r._2}"
+      ConfigurableScenario(name, scenario(name).exec(Pages.APIs.domainSearch(t)), r._1, r._2)
+    }
+  }
+
+  val configuredScenarios: List[ConfigurableScenario] = List(
+    ConfigurableScenario(HomePage.name, homePage, 10, 10 seconds),
+    ConfigurableScenario(UserStorySmoking.name, searchSmoke, 10, 10 seconds)
+  ) ++ domainSearchApi
```
Comment (calbach) [inline performance/src/gatling/simulations/org/pmiops/databrowser/Scenarios.scala]: `++` is a binary operator in Scala? :slightly_frowning_face:

**Review by calbach — APPROVED** [review]
Looks good thanks, just a doc nit

File: performance/src/gatling/simulations/org/pmiops/databrowser/Scenarios.scala (hunk 2 above)
Comment (rushtong) [reply performance/src/gatling/simulations/org/pmiops/databrowser/Scenarios.scala]: It's just list concatenation here. `:::` is actually more idiomatic, but both `++` and `:::` let you do something that is not great - they let you mix types which can lead to unexpected type erasure problems. 

Expected use:
```
scala> List("a", "b") ++ List("c", "d")
res0: List[String] = List(a, b, c, d)
```
Can be confusing:
```
scala> List("a", "b") ++ List(1, 2)
res1: List[Any] = List(a, b, 1, 2)

scala> List("a", "b") ::: List(1, 2)
res2: List[Any] = List(a, b, 1, 2)
```
~Typesafe using `List.concat`:~
```
scala> List.concat(("c", "d"), (1, 2))
<console>:12: error: type mismatch;
 found   : (String, String)
 required: Traversable[?]
       List.concat(("c", "d"), (1, 2))
                   ^
<console>:12: error: type mismatch;
 found   : (Int, Int)
 required: Traversable[?]
       List.concat(("c", "d"), (1, 2))
                               ^
```

~So, thanks for asking - I'll use concat to be more accurate about what I'm expecting here.~

^^^ is wrong, sorry. Concat lets you do the same unsafe thing. 
```
scala> List.concat(List("c", "d"), List(1, 2))
res0: List[Any] = List(c, d, 1, 2)
```

File: performance/src/gatling/simulations/org/pmiops/databrowser/Scenarios.scala (hunk 1 above)
Comment (rushtong) [reply performance/src/gatling/simulations/org/pmiops/databrowser/Scenarios.scala]: Apparently not ... thank you!

**Comment by rushtong** [comment]
@calbach - PTAL, I fixed some minor codacy warnings - all legit. Nothing structural has changed - hopefully it's a little clearer.

**Review by calbach — APPROVED** [review]
Looks good - thanks

**Pull request merged**